sdk-extensions/aws/src/main/java/io/opentelemetry/sdk/extension/aws/trace/SamplingRuleApplier.java

```diff
@@ -50,7 +57,21 @@
   private final Matcher serviceTypeMatcher;
   private final Matcher resourceArnMatcher;
 
-  SamplingRuleApplier(GetSamplingRulesResponse.SamplingRule rule) {
+  private final Statistics statistics;
+
+  SamplingRuleApplier(String clientId, GetSamplingRulesResponse.SamplingRule rule) {
+    this.clientId = clientId;
+    ruleName = rule.getRuleName();
+    if (rule.getReservoirSize() > 0) {
+      // Until calling GetSamplingTargets, the default is to borrow 1/s if reservoir size is
+      // positive.
+      reservoirSampler = new RateLimitingSampler(1);
+      borrowing = true;
```

Comment:
In a separate (and hopefully final) PR, I will add support for "updating" a rule based on a sampling target - what X-Ray returns in real time based on the statistics collected from all client IDs (processes). That will be the case where reservoir sampler is set but borrowing is false.